files/check_network_shares

```
@@ -0,0 +1,33 @@
+#!/bin/sh
+
+set -e
+cd / || exit 1
+
+fail() {
+  printf "1"
+}
+
+# remove comments; remove blank lines; normalize spacing to a single space
+# select mount point and filesystem
+# ensure expected filesystem matches actual filesystem for network filesystems
+# timeout prevents a stale NFS share from freezing the check
+
+# Skips "noauto" filesystems, but will also skip "noauto_da_alloc". As
+# "noauto_da_alloc" applies only to ext4, this won't be a problem in practice.
+
+# Due to the use of pipes, the while opperates in a subshell, so there
+# is no communication to outside the loop. Each failed share will prepend
+# a "1" to the 0, still providing a value greater than 0 and can allow
+# something more sophisticated to count the number of failed shares.
```

Are we configuring Zabbix to check the exit code? Or the value returned? I tested a simple umount on p-u-ansible01 and it doesn't seem to be providing the correct exit code. Wondering if fail() should be `exit 1` instead of `print "1"`

```
[avuong@p-u-ansible01 ~]$ sudo mount /data
[avuong@p-u-ansible01 ~]$ ./checkmount.sh
0
[avuong@p-u-ansible01 ~]$ echo $?
0
[avuong@p-u-ansible01 ~]$ sudo umount /data
[avuong@p-u-ansible01 ~]$ ./checkmount.sh
10
[avuong@p-u-ansible01 ~]$ echo $?
0
```

---

Here's another test with a CIFS and NFS mount on the same server:

```
[root@p-w-imagegen01 avuong]# cat /etc/fstab | egrep "nfs|cifs" | grep -v "#"
svm-netapp-dlib.in.library.ucla.edu:/Masters/dlmasters/latimes/masters /image-gen/latimes nfs vers=3,_netdev,intr,hard,nosuid,noexec 0 0
//svm-netapp-dlib.in.library.ucla.edu/Masters /mnt/ucla/masters cifs credentials=/root/.emeluser-cifs,vers=3.0 0 0
[root@p-w-imagegen01 avuong]# df -Th | egrep "nfs|cifs"
svm-netapp-dlib.in.library.ucla.edu:/Masters/dlmasters/latimes/masters nfs        38T   31T  7.8T  80% /image-gen/latimes
//svm-netapp-dlib.in.library.ucla.edu/Masters                          cifs      973M  1.1M  972M   1% /mnt/ucla/masters
[root@p-w-imagegen01 avuong]# ./checkmount.sh
10
[root@p-w-imagegen01 avuong]# echo $?
0
[root@p-w-imagegen01 avuong]# umount /mnt/ucla/masters
[root@p-w-imagegen01 avuong]# ./checkmount.sh
10
[root@p-w-imagegen01 avuong]# echo $?
0
[root@p-w-imagegen01 avuong]# df -Th | egrep "nfs|cifs"
svm-netapp-dlib.in.library.ucla.edu:/Masters/dlmasters/latimes/masters nfs        38T   31T  7.8T  80% /image-gen/latimes
```

I guess I was expecting an exit code of 1 to be returned after the umount.

---

Zabbix relies upon stdout/stderr.

---

It looks like there's support for exit codes:

https://www.zabbix.com/documentation/5.0/manual/appendix/command_execution

I'm not that opinionated on the check, just wondering what values zabbix would be checking.